### PR TITLE
Update all dependencies

### PR DESCRIPTION
**Pull request by chapmanc**
Most of the dependencies were out of date so this PR updates all of them. 

This has been used in the load test and is a known good configuration.

### CODE REVIEW

File: internal/otel/config/helpers/receivers/prometheus_receiver.go [hunk 1]
```diff
@@ -40,18 +40,18 @@ func PrometheusReceiverCfg() *PrometheusConfig {
...
 	*/
 
 	return &PrometheusConfig{
 		Config: map[string][]ScrapeConfig{
-			scrapeConfigKey: []ScrapeConfig{
+			scrapeConfigKey: {
 				{
 					JobName:        "consul-telemetry-collector",
-					ScrapeInterval: "10s",
+					ScrapeInterval: "1m",
```
Comment (Achooo) [inline internal/otel/config/helpers/receivers/prometheus_receiver.go:54]: This is to make it the same as Consul ? That makes sense to me 👍🏽

File: receivers/envoyreceiver/envoy.go [hunk 2]
```diff
@@ -85,8 +86,14 @@ func (r *envoyReceiver) Start(_ context.Context, host component.Host) error {
 }
 
 func (r *envoyReceiver) Shutdown(_ context.Context) error {
-	r.grpcServer.GracefulStop()
-	<-r.shutdownCh
+	if r.grpcServer != nil {
```
Comment (Achooo) [inline receivers/envoyreceiver/envoy.go:89]: Early return instead?

```suggestion
	if r.grpcServer == nil { 
	    r.logger.Warn("Shutting down envoy receiver that did not start successfully")
	    return
	}

	r.logger.Info("Shutting down envoy receiver gracefully")
	r.grpcServer.GracefulStop()
	<-r.shutdownCh
```

File: internal/otel/config/helpers/receivers/prometheus_receiver.go (hunk 1 above)
Comment (chapmanc) [reply internal/otel/config/helpers/receivers/prometheus_receiver.go:54]: Kind of ... it's not going to be 1m as the collector will send as soon as it's "bucket" of metrics is full. This is more to scrape itself less.

**Review by Achooo — APPROVED** [review]

File: internal/otel/config/helpers/receivers/prometheus_receiver.go (hunk 1 above)
Comment (Achooo) [reply internal/otel/config/helpers/receivers/prometheus_receiver.go:54]: Ah sounds good, thanks for the context 👍🏽

**Review by clly — APPROVED** [review]

**Pull request merged**